.github/workflows/benchmark-releases.yaml

```diff
@@ -0,0 +1,47 @@
+---
+name: Benchmarks nightly between last release and current main on AMD64
+permissions: read-all
+on:
+  schedule:
+    - cron: '25 5 * * *' # runs every day at 09:25 UTC
+  # workflow_dispatch enables manual testing of this job by maintainers
+  workflow_dispatch:
+jobs:
+  benchmark-pull-request:
+    runs-on: ubuntu-latest-8-cores
+    steps:
+    - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+      with:
+        fetch-depth: 0
+    - id: goversion
+      run: echo "goversion=$(cat .go-version)" >> "$GITHUB_OUTPUT"
+    - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+      with:
+        go-version: ${{ steps.goversion.outputs.goversion }}
+    - name: Run Benchmarks
```

Comment:
maybe worth sharing the benchmark as a workflow template?

Reply:
Thanks, @tjungblu! I wanted to deliver this, but I've been busy with other tasks. I would suggest (following your own suggestion ;) adding it to a template to keep it DRY.

Reply:
I'm also out, and will be back on Monday :see_no_evil:. But I also think there's no rush to deliver this.

Reply:
just refactored it, if you have a brief couple minutes to review - would be much appreciated @ivanvc